openlibrary/plugins/upstream/addbook.py

```diff
@@ -1009,11 +1009,20 @@ def process_input(self, i):
         if 'author' in i:
             author = trim_doc(i.author)
             alternate_names = author.get('alternate_names', None) or ''
-            author.alternate_names = [
+            alternate_names = [
                 name.strip()
                 for name in alternate_names.replace("\n", ";").split(';')
                 if name.strip()
             ]
+            author.alternate_names = []
+            seen = {}
+            for name in alternate_names:
+                caseinsensitive = name.lower()
+                if caseinsensitive in seen:
+                    continue
+                seen[caseinsensitive] = 1
+                author.alternate_names.append(name)
```

Comment:
You can replace this with a call to the helper fn `uniq` ( openlibrary.core.helpers ) ; it takes an optional `key` parameter that transforms each item before deduping.

Reply:
Done, also rebased on #8366 as it touches the line of code immediately above this, to change the delimeter.